### PR TITLE
Nova 2.9.0 compatibility.

### PR DESCRIPTION
**Pull request by vbezruchkin**

### CODE REVIEW

**Comment by dkulyk** [comment]
Fields methods should return FieldCollection. 

$resource does not used in rulesForUpdate

**Comment by vbezruchkin** [comment]
They do return FieldCollection. Probably I missed something in specific cases, but it worked well for me. The key change is to switch collect to new FieldCreation object.

rulesForUpdate updated to match the new version of Laravel Nova which has two params now.

**Comment by bajramemini** [comment]
Thanks @vbezruchkin 

Will try this out and tag a new release asap.

Regards

File: src/TabsOnEdit.php [hunk 1]
```diff
@@ -15,11 +16,11 @@ trait TabsOnEdit
      */
     public function creationFields(NovaRequest $request)
     {
-        return new FieldCollection(
+        return collect(
             [
                 'Tabs' => [
                     'component' => 'tabs',
-                    'fields'    => $this->removeNonCreationFields($request, $this->resolveFields($request)),
+                    'fields'    => $this->removeNonCreationFields($this->resolveFields($request)),
```
Comment (dkulyk) [inline src/TabsOnEdit.php]:
```php
    /**
     * Remove non-creation fields from the given collection.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @param  \Laravel\Nova\Fields\FieldCollection  $fields
     * @return \Laravel\Nova\Fields\FieldCollection
     */
    protected function removeNonCreationFields(NovaRequest $request, FieldCollection $fields);
```

File: src/TabsOnEdit.php [hunk 2]
```diff
@@ -15,11 +16,11 @@ trait TabsOnEdit
      */
     public function creationFields(NovaRequest $request)
```
Comment (dkulyk) [inline src/TabsOnEdit.php]:
```php
    /**
     * Resolve the creation fields.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @return \Laravel\Nova\Fields\FieldCollection
     */
    public function creationFields(NovaRequest $request);
```

File: src/TabsOnEdit.php [hunk 3]
```diff
@@ -88,7 +89,7 @@ public static function rulesForCreation(NovaRequest $request)
      * @param  \Laravel\Nova\Http\Requests\NovaRequest $request
      * @return array
      */
-    public static function rulesForUpdate(NovaRequest $request)
+    public static function rulesForUpdate(NovaRequest $request, $resource = null)
     {
         return static::formatRules($request, (new static(static::newModel()))
```
Comment (dkulyk) [inline src/TabsOnEdit.php]:
```php
    public static function rulesForUpdate(NovaRequest $request, $resource = null)
    {
        $resource = $resource ?? new static(static::newModel());

        return static::formatRules($request, $resource->updateFields($request)
```

**Pull request merged**